common/libs/units/units.h

```diff
@@ -110,11 +110,31 @@ template <class Q, class ValTy> class Scalar {
   ValTy val_;
 };
 
-// Represents a Scalar where addition and subtraction are well-defined.
+// Represents a Scalar where:
+// - addition and subtraction are well-defined
+// - multiplication by an external factor of type ValTy is defined
```

Comment:
Modules don't (necessarily) have division, right?  What's the more precise mathematical term?

Reply:
Semigroup ring maybe? We shouldn't get too carried away here, let's just say it resembles a module and describe the whole thing in plain language.

Reply:
Indeed, it only has division if the external Ring you use to build your Module's external law is a Division Ring (division is multiplication by the element's inverse). Not sure there is a specific word for this.
But it is kind of still the case here... if we were to build a UnitScalar<TestScalar, int>, the division would not be properly defined, and the / operator would not work.
This probably needs addressing, at least in a comment (see your above remark on comments).

Reply:
I'll remove the "module" comment anyway, as it was only here to justify the name of the ModuleScalar template, which disappeared.
And add the notion that division has to be defined on ValTy.